indica/workflows/abstract_bayes_workflow.py

```diff
@@ -150,330 +46,270 @@ def _build_result_dict(self):
+                for num_imp, imp in enumerate(self.plasma_context.plasma_settings.impurities)
             },
         }
         result["PROFILE_STAT"] = {
             **result["PROFILE_STAT"],
             **{
                 f"TIZ{num_imp + 1}": self.blobs["ion_temperature"].sel(element=imp)
-                for num_imp, imp in enumerate(self.plasma.impurities)
+                for num_imp, imp in enumerate(self.plasma_context.plasma_settings.impurities)
             },
         }
 
         result["OPTIMISATION"] = {
-            "ACCEPT_FRAC": self.accept_frac,
-            "PRIOR_SAMPLE": self.prior_sample,
-            "POST_SAMPLE": self.post_sample,
-            "AUTOCORR": self.autocorr,
+            "ACCEPT_FRAC": self.opt_samples["accept_frac"],
+            "PRIOR_SAMPLE": self.opt_samples["prior_sample"],
+            "POST_SAMPLE": self.opt_samples["post_sample"],
+            "AUTO_CORR": self.opt_samples["auto_corr"],
+            # "GELMAN_RUBIN": gelman_rubin(self.sampler.get_chain(flat=False))
         }
 
         result["GLOBAL"] = {
             "TI0": self.blobs["ion_temperature"]
-            .sel(element=self.plasma.main_ion)
-            .sel(rho_poloidal=0, method="nearest")
-            .median(dim="index"),
+                .sel(element=self.plasma_context.plasma_settings.main_ion)
+                .sel(rho_poloidal=0, method="nearest")
+                .median(dim="index"),
             "TE0": self.blobs["electron_temperature"]
-            .sel(rho_poloidal=0, method="nearest")
-            .median(dim="index"),
+                .sel(rho_poloidal=0, method="nearest")
+                .median(dim="index"),
             "NE0": self.blobs["electron_density"]
-            .sel(rho_poloidal=0, method="nearest")
-            .median(dim="index"),
+                .sel(rho_poloidal=0, method="nearest")
+                .median(dim="index"),
             "NI0": self.blobs["ion_density"]
-            .sel(element=self.plasma.main_ion)
-            .sel(rho_poloidal=0, method="nearest")
-            .median(dim="index"),
-            "TI0_ERR": self.blobs["ion_temperature"]
-            .sel(element=self.plasma.main_ion)
-            .sel(rho_poloidal=0, method="nearest")
-            .std(dim="index"),
-            "TE0_ERR": self.blobs["electron_temperature"]
-            .sel(rho_poloidal=0, method="nearest")
-            .std(dim="index"),
-            "NE0_ERR": self.blobs["electron_density"]
-            .sel(rho_poloidal=0, method="nearest")
-            .std(dim="index"),
-            "NI0_ERR": self.blobs["ion_density"]
-            .sel(element=self.plasma.main_ion)
-            .sel(rho_poloidal=0, method="nearest")
-            .std(dim="index"),
+                .sel(element=self.plasma_context.plasma_settings.main_ion)
+                .sel(rho_poloidal=0, method="nearest")
+                .median(dim="index"),
+            "WP": self.blobs["wp"]
```

Comment:
why median and not mean?

Reply:
If you have a skewed distribution the mean might be misleading. I feel it is more intuitive to think of Bayesian inference in terms of quantiles i.e. median, 25-75% etc. I'm curious if you have a reason for preferring the mean?